core/src/Xios.cpp

```diff
@@ -213,69 +275,517 @@ void Xios::setCalendarTimestep(cxios_duration timestep)
+    int start;
+    cxios_get_domain_ibegin(domain, &start);
+    return start;
+}
+
+/*!
+ * Get the local starting latitude for a given domain
+ *
+ * @param the domain ID
+ * @return the local starting latitude of the corresponding domain
+ */
+int Xios::getDomainLatitudeStart(std::string domainId)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    int start;
+    cxios_get_domain_jbegin(domain, &start);
+    return start;
+}
+
+/*!
+ * Get the local longitude values for a given domain
+ *
+ * @param the domain ID
+ * @return the local longitude values of the corresponding domain
+ */
+std::vector<double> Xios::getDomainLongitudeValues(std::string domainId)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    int size = getDomainLongitudeSize(domainId);
+    double* values = new double[size];
+    cxios_get_domain_lonvalue_1d(domain, values, &size);
+    std::vector<double> vec(values, values + size);
+    delete[] values;
+    return vec;
+}
+
+/*!
+ * Get the local latitude values for a given domain
+ *
+ * @param the domain ID
+ * @return the local latitude values of the corresponding domain
+ */
+std::vector<double> Xios::getDomainLatitudeValues(std::string domainId)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    int size = getDomainLatitudeSize(domainId);
+    double* values = new double[size];
+    cxios_get_domain_latvalue_1d(domain, values, &size);
+    std::vector<double> vec(values, values + size);
+    delete[] values;
+    return vec;
+}
+
+/*!
+ * Get the field associated with a given ID
+ *
+ * @param the field ID
+ * @return a pointer to the XIOS CField object
+ */
+xios::CField* Xios::getField(std::string fieldId)
+{
+    xios::CField* field = NULL;
+    cxios_field_handle_create(&field, fieldId.c_str(), fieldId.length());
+    return field;
+}
+
+/*!
+ * Get the name of a field with a given ID
+ *
+ * @param the field ID
+ * @return name of the corresponding field
+ */
+std::string Xios::getFieldName(std::string fieldId)
+{
+    int size = 20;
+    char cStr[size];
+    xios::CField* field = getField(fieldId);
+    cxios_get_field_name(field, cStr, size);
+    std::string fieldName(cStr, size);
+    boost::algorithm::trim_right(fieldName);
+    return fieldName;
+}
+
+/*!
+ * Get the operation associated with a field with a given ID
+ *
+ * @param the field ID
+ * @return operation used for the corresponding field
+ */
+std::string Xios::getFieldOperation(std::string fieldId)
+{
+    int size = 20;
+    char cStr[size];
+    xios::CField* field = getField(fieldId);
+    cxios_get_field_operation(field, cStr, size);
+    std::string operation(cStr, size);
+    boost::algorithm::trim_right(operation);
+    return operation;
+}
+
+/*!
+ * Get the grid reference associated with a field with a given ID
+ *
+ * @param the field ID
+ * @return grid reference used for the corresponding field
+ */
+std::string Xios::getFieldGridRef(std::string fieldId)
+{
+    int size = 20;
```

Comment:
We may want to make `size` a class variable so we can change it once for all fields? Possibly it could become a user-defined input through the `Nextsim::Configurator`

Reply:
Good idea. Addressed in f60e88fc19cc50ff0cbde3b0438daa6798b5d6ef, just adding a public attribute for now. But yes, could be added to the configurator later, too.